### ISSUE TITLE
CoreOS Container Linux FastPatch Updates

### ISSUE THREAD

**Issue by gardener-robot-ci-1** [gardener/gardener#22]
**Issue by [kubernetes-jenkins](https://github.com/kubernetes-jenkins)**
_Wednesday Jul 19, 2017 at 18:45 GMT_
_Originally opened as https://git.removed/kubernetes-attic/garden-operator/issues/37_

----

# Story
* As operator I want OS patches (functional and security) to be deployed as fast and automated as possible and ideally without any significant downtime, so that my cluster remains sound and safe. 

# Motivation
While we can do rolling updates on IaaS hyperscale providers (for now), we shouldnt do that respectively waste too much time doing it on bare metal.

# Acceptance Criteria
- [ ] Update shoot cluster worker nodes using FastPatch (a.k.a. a/b channel updates without replacing the VM)
  - [ ] Shoot cluster TPR should hold exact garden operator version (which includes configuration such as CoreOS Container Linux image tag and version)
  - [ ] When a new garden operator gets deployed, it checks existing shoot clusters and updates them according to a canary process
    - [ ] At first only one cluster is updated, validating that it properly works after update
    - [ ] Then all other clusters are updated in parallel according to a max_in_flight property

# Definition of Done

- [ ] Knowledge is distributed: Have you spread your knowledge in pair programming/code review?
- [ ] Unit Tests are provided: Have you written automated unit tests or added manual NGPTT tickets?
- [ ] Integration Tests are provided: Have you written automated integration tests?
- [ ] Minimum API exposure: If you have added public API, was it really necessary/is it minimal?
- [ ] Operations guide: Have you updated the [operations guide](https://git.infra.removed/kubernetes/kube-docs/wiki/Operations%20Guide)?

:information_source: Migrated from Jira issue [KUBE-170](https://jtrack.removed/browse/KUBE-170)

**Comment by gardener-robot-ci-1** [comment]
**Comment by [mvladev](https://github.com/mvladev)**
_Saturday Aug 19, 2017 at 09:10 GMT_

----

CoreOS created an [core-linux-update-operator](https://github.com/coreos/container-linux-update-operator) for that purpose. I think its worth investigating.

**Comment by vlerenc** [comment]
We won't do them now that we can easily replace entire VMs.
